### PR TITLE
[FLINK-6442] [table] Extend TableAPI Support Sink Table Registration …

### PR DESCRIPTION
**Pull request by lincoln-lil**
…and ‘insert into’ Clause in SQL

Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [ ] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [ ] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [ ] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/utils/CsvSQLTableSink.scala [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import org.apache.flink.api.common.typeinfo.TypeInformation
+import org.apache.flink.api.java.typeutils.RowTypeInfo
+import org.apache.flink.core.fs.FileSystem.WriteMode
+import org.apache.flink.table.sinks.{CsvTableSink, TableSink}
+import org.apache.flink.types.Row
+
+class CsvSQLTableSink(
+    fieldTypes: Array[TypeInformation[_]],
+    fieldNames: Option[Array[String]],
+    path: String,
```
Comment (hequn8128) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/utils/CsvSQLTableSink.scala]: 把Option的参数都统一放最后是不是要好点(path放最前面)，构造函数如果不写fieldNames参数变成两个了

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala [hunk 2]
```diff
@@ -464,21 +477,52 @@ abstract class TableEnvironment(val config: TableConfig) {
...
+      val insert = parsed.asInstanceOf[SqlInsert]
+      // validate sink table
+      val targetName = insert.getTargetTable.asInstanceOf[SqlIdentifier].names.get(0)
+      val targetTable = getTable(targetName)
+      if (null == targetTable || !targetTable.isInstanceOf[TableSinkTable[_]]) {
+        throw new TableException("SQL DML INSERT operation need a registered TableSink Table!")
+      }
+      // validate unsupported partial insertion to sink table
+      val sinkTable = targetTable.asInstanceOf[TableSinkTable[_]]
+      if (null != insert.getTargetColumnList && insert.getTargetColumnList.size() !=
```
Comment (hequn8128) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala]: rowType是不是也要一样

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/batch/sql/TableWithSQLITCase.scala [hunk 3]
```diff
@@ -113,4 +116,52 @@ class TableWithSQLITCase(
...
+    val sql = "INSERT INTO targetTable SELECT a, b, c FROM sourceTable"
+    tEnv.sql(sql)
+    env.execute()
+
+    val expected = Seq("1,1,Hi", "2,2,Hello", "3,2,Hello world").mkString("\n")
+    TestBaseUtils.compareResultsByLinesInMemory(expected, path)
+  }
+
+  /** test unsupported partial insert **/
+  @Test(expected = classOf[TableException])
```
Comment (hequn8128) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/batch/sql/TableWithSQLITCase.scala]: 异常的test case放在继承TableTestBase的某个类下

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/stream/sql/SqlITCase.scala [hunk 4]
```diff
@@ -1140,6 +1144,53 @@ class SqlITCase extends StreamingWithStateTestBase {
...
+    val sql = "INSERT INTO targetTable SELECT a, b, c FROM sourceTable"
+    tEnv.sql(sql)
+    env.execute()
+
+    val expected = Seq("1,1,Hi", "2,2,Hello", "3,2,Hello world").mkString("\n")
+    TestBaseUtils.compareResultsByLinesInMemory(expected, path)
+  }
+
+  /** test unsupported partial insert **/
+  @Test(expected = classOf[TableException])
```
Comment (hequn8128) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/stream/sql/SqlITCase.scala]: 异常的test case放在继承TableTestBase的某个类下

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala (hunk 2 above)
Comment (lincoln-lil) [reply flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala]: rowtype 在sql validate那一层已经验证过了

**Pull request closed without merging**